app/helpers/react_on_rails_helper.rb

```diff
@@ -102,11 +104,13 @@ def react_component(component_name, raw_options = {})
     # The reason is that React is smart about not doing extra work if the server rendering did its job.
 
     component_specification_tag =
-      content_tag(:div,
-                  "",
+      content_tag(:script,
                   class: "js-react-on-rails-component",
-                  style: options.style,
-                  data: options.data)
+                  style: nil,
+                  data: options.data) do
+        props = Yajl.dump(options.props.is_a?(String) ? JSON.parse(options.props) : options.props)
+        "var #{options.dom_id.tr('-', '_')} = #{props};".html_safe
```

Comment:
I would suggest not playing games with global Javascript variables but instead use

````
<script type="application/json" id="...">{ "json": "encoded-data" }</script>
````

Also, it would really help development if you use human-readable JSON in non-development environment, a-la:

````
  def json_safe_and_pretty(something)
    if Rails.env.development?
      json_escape(JSON.pretty_generate(something.as_json))
    else
      json_escape(something.to_json)
    end
  end
````